### PR TITLE
Update the main energy datasets

### PR DESCRIPTION
**Pull request by pabloarosado**
This PR:
* Refactored many energy steps (not all), to adapt to the new ETL changes.
* Improved metadata, adapting to the new fields.

Reviewers:
There's no need to have a thorough review, since we'll have a bug bash. So feel free to just skim through it to see if you spot anything weird. 

* @Marigold would you mind reviewing the code (`.py` files)? 
* @spoonerf do you mind reviewing the metadata files?

To make the bug bash easier, I'll merge to master tomorrow morning, but please feel free to review it even after merging. Again, no need to go through every file in detail.

Thanks a lot!

### CODE REVIEW

File: etl/steps/data/garden/papers/2023-12-12/nemet_2009.meta.yml [hunk 1]
```diff
@@ -1,2 +1,24 @@
...
-  title: Cost and capacity of photovoltaic technology (Nemet, 2009)
+  title: Cost and capacity of photovoltaic technology
+  update_period_days: 0
+
+tables:
+  nemet_2009:
+    variables:
+      cost:
+        title: Cost of photovoltaic technology
+        units: 2004 USD/Watt
```
Comment (Marigold) [inline etl/steps/data/garden/papers/2023-12-12/nemet_2009.meta.yml:18]: Should be `unit`, not `units`.

File: snapshots/eia/2023-12-12/international_energy_data.zip.dvc [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
+    # description: |-
+    # attribution:
+    attribution_short: EIA
+    citation_full: |-
+      U.S. Energy Information Administration (EIA) - International Energy Data (2023).
+    date_accessed: 2023-12-12
+    date_published: 2023-08-01
+    url_main: https://www.eia.gov/opendata/bulkfiles.php
+    url_download: https://api.eia.gov/bulk/INTL.zip
+  license:
```
Comment (Marigold) [inline snapshots/eia/2023-12-12/international_energy_data.zip.dvc:14]: license should be nested under `origin`

File: snapshots/energy_institute/2023-12-12/statistical_review_of_world_energy.csv.dvc [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
+    attribution: Energy Institute - Statistical Review of World Energy (2023)
+    attribution_short: Energy Institute
+    citation_full: |-
+      Energy Institute - Statistical Review of World Energy (2023).
+    date_accessed: 2023-12-12
+    date_published: 2023-06-26
+    url_main: https://www.energyinst.org/statistical-review/
+    url_download:
+      https://www.energyinst.org/__data/assets/file/0007/1055761/Consolidated-Dataset-Panel-format-CSV.csv
+  license:
```
Comment (Marigold) [inline snapshots/energy_institute/2023-12-12/statistical_review_of_world_energy.csv.dvc:18]: license should be nested under origin

File: snapshots/energy_institute/2023-12-12/statistical_review_of_world_energy.xlsx.dvc [hunk 4]
```diff
@@ -1,22 +1,25 @@
...
+    attribution: Energy Institute - Statistical Review of World Energy (2023)
+    attribution_short: Energy Institute
+    citation_full: |-
+      Energy Institute - Statistical Review of World Energy (2023).
+    date_accessed: 2023-12-12
+    date_published: 2023-06-26
+    url_main: https://www.energyinst.org/statistical-review/
+    url_download:
+      https://www.energyinst.org/__data/assets/excel_doc/0007/1055545/EI-stats-review-all-data.xlsx
+  license:
```
Comment (Marigold) [inline snapshots/energy_institute/2023-12-12/statistical_review_of_world_energy.xlsx.dvc:17]: Should be under origin

**Review by Marigold — APPROVED** [review]
Holy crap, 4000 lines 😅 It looks solid though. Ignore the walden unit tests failure, should be fixed on master. (I also fixed datadiff bug, but it'd be nice to make it useful for changes like this)

File: etl/steps/data/garden/papers/2023-12-12/nemet_2009.meta.yml (hunk 1 above)
Comment (pabloarosado) [reply etl/steps/data/garden/papers/2023-12-12/nemet_2009.meta.yml:18]: Ouch!! In my new laptop I had not yet properly configured the path to the yaml definitions schema. Now I see it as an issue. Thanks for pointing it out, fixed!

File: etl/steps/data/garden/papers/2023-12-12/smil_2017.yml [hunk 5]
```diff
@@ -1,43 +1,37 @@
+definitions:
+  common:
```
Comment (spoonerf) [inline etl/steps/data/garden/papers/2023-12-12/smil_2017.yml:2]: The filename here is missing 'meta', I think that will prevent it being used?

File: etl/steps/data/garden/energy/2023-12-12/energy_mix.meta.yml [hunk 6]
```diff
@@ -1,25 +1,1266 @@
 definitions:
   common:
+    processing_level: major
     presentation:
       topic_tags:
-      - Energy mix
+      - "Energy"
```
Comment (spoonerf) [inline etl/steps/data/garden/energy/2023-12-12/energy_mix.meta.yml:6]: I don't know if it matters that the topic here has quotes? I just noticed it as being unusual

**Review by spoonerf — COMMENTED** [review]
Looks good - just a couple of tiny comments!

File: etl/steps/data/garden/papers/2023-12-12/smil_2017.yml (hunk 5 above)
Comment (pabloarosado) [reply etl/steps/data/garden/papers/2023-12-12/smil_2017.yml:2]: Good catch, indeed, the metadata was ignored!

**Pull request merged**